### PR TITLE
8311663: Additional refactoring of Locale tests to JUnit

### PR DESCRIPTION
**Pull request by justin-curtis-lu**
Please review this PR which refactors more java.util.Locale tests to JUnit with some minor cleanup as well.

Although some of the files could benefit from being renamed bugNNNNNNN to something more descriptive, this makes reviewing harder, and will be handled separately.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8311663](https://bugs.openjdk.org/browse/JDK-8311663): Additional refactoring of Locale tests to JUnit (**Sub-task** - P4)

### Reviewers
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14881/head:pull/14881` \
`$ git checkout pull/14881`

Update a local copy of the PR: \
`$ git checkout pull/14881` \
`$ git pull https://git.openjdk.org/jdk.git pull/14881/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14881`

View PR using the GUI difftool: \
`$ git pr show -t 14881`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14881.diff">https://git.openjdk.org/jdk/pull/14881.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14881#issuecomment-1635054361)

### CODE REVIEW

File: test/jdk/java/util/Locale/Bug8135061.java [hunk 1]
```diff
@@ -35,47 +35,38 @@
...
+     * as the matching tag.
+     */
+    @Test
+    public void lookupReturnValueTest() {
+        List<LanguageRange> ranges = LanguageRange.parse("i-navajo");
+        Collection<Locale> locales = Collections.singleton(Locale.of("nv"));
+        Locale match = Locale.lookup(ranges, locales);
+        assertEquals(match.toLanguageTag(), "nv", "[Locale.lookup failed on language"
+                + " range: " + ranges + " and language tags "
+                + locales + "]");
```
Comment (naotoj) [inline test/jdk/java/util/Locale/Bug8135061.java:70]: Same as above

File: test/jdk/java/util/Locale/Bug8159420.java [hunk 2]
```diff
@@ -32,119 +32,97 @@
...
-
-        } finally {
-            Locale.setDefault(origLocale);
-        }
+public class Bug8159420 {
 
+    // Ensure all tests run with a Turkish Locale
+    @BeforeAll
+    static void setTurkishLocale() {
+        Locale.setDefault(Locale.of("tr", "TR"));
```
Comment (naotoj) [inline test/jdk/java/util/Locale/Bug8159420.java:67]: Should the tests run under `othervm` mode?

File: test/jdk/java/util/Locale/Bug8179071.java [hunk 3]
```diff
@@ -38,53 +38,60 @@
...
-        shortJanuaryNames.forEach(Bug8179071::test);
-        checkInvalidTags();
+    // getAvailableLocales() should not contain any deprecated or Legacy language tags
+    @Test
+    public void invalidTagsTest() {
+        Set<String> invalidTags = new HashSet<>();
+        Arrays.stream(Locale.getAvailableLocales())
+                .map(Locale::toLanguageTag)
+                .forEach(tag -> {if(LegacyAliases.contains(tag)) {invalidTags.add(tag);}});
+        assertEquals(true, invalidTags.isEmpty(),
```
Comment (naotoj) [inline test/jdk/java/util/Locale/Bug8179071.java:94]: `assertTrue()` may fit better here

File: test/jdk/java/util/Locale/ThaiGov.java [hunk 4]
```diff
@@ -20,98 +20,70 @@
...
 
-        if (!strExpected.equals(str)) {
-            throw new RuntimeException();
-        }
+    private static final double VALUE = 12345678.234;
+    private static final Locale TH = Locale.of("th", "TH", "TH");
 
+    // Test number formatting for thai
+    @Test
+    public void numberTest() throws RuntimeException {
```
Comment (naotoj) [inline test/jdk/java/util/Locale/ThaiGov.java:51]: Not your change, but this `throws` is redundant

**Review by naotoj — COMMENTED** [review]
Thanks for the conversion, Justin. Looks good overall. Some minor comments follow.

File: test/jdk/java/util/Locale/Bug8179071.java [hunk 5]
```diff
@@ -38,53 +38,60 @@
...
-                .map(loc -> loc.toLanguageTag())
-                .forEach( tag -> {if(LegacyAliases.contains(tag)) {invalidTags.add(tag);}});
-        if (!invalidTags.isEmpty()) {
-          throw new RuntimeException("failed: Deprecated and Legacy tags found  " + invalidTags  + " in AvailableLocales ");
-        }
+    // Expected month format data for locales after language aliases replacement.
+    private static Stream<Arguments> shortJanuaryNames() {
+        return Stream.of(
+                Arguments.of("pa-PK", "\u0a1c\u0a28"),
+                Arguments.of("uz-AF" , "yan"),
```
Comment (turbanoff) [inline test/jdk/java/util/Locale/Bug8179071.java:78]:
```suggestion
                Arguments.of("uz-AF", "yan"),
```

File: test/jdk/java/util/Locale/Bug8179071.java [hunk 6]
```diff
@@ -38,53 +38,60 @@
...
-        checkInvalidTags();
+    // getAvailableLocales() should not contain any deprecated or Legacy language tags
+    @Test
+    public void invalidTagsTest() {
+        Set<String> invalidTags = new HashSet<>();
+        Arrays.stream(Locale.getAvailableLocales())
+                .map(Locale::toLanguageTag)
+                .forEach(tag -> {if(LegacyAliases.contains(tag)) {invalidTags.add(tag);}});
+        assertEquals(true, invalidTags.isEmpty(),
+                "Deprecated and Legacy tags found  " + invalidTags  + " in AvailableLocales ");
```
Comment (turbanoff) [inline test/jdk/java/util/Locale/Bug8179071.java:95]:
```suggestion
                "Deprecated and Legacy tags found  " + invalidTags + " in AvailableLocales ");
```

Comment (justin-curtis-lu) [reply test/jdk/java/util/Locale/Bug8179071.java]: Thanks, updated this and the other suggested WS change

File: test/jdk/java/util/Locale/ThaiGov.java (hunk 4 above)
Comment (justin-curtis-lu) [reply test/jdk/java/util/Locale/ThaiGov.java]: Thanks for the review, addressed this and your other suggestions as well.

File: test/jdk/java/util/Locale/Bug8135061.java [hunk 7]
```diff
@@ -35,47 +35,38 @@
...
-        } catch (Exception ex) {
-            throw new RuntimeException("[Locale.lookup failed on language"
-                    + " range: " + ranges + " and language tags "
-                    + locales + "]", ex);
-        }
-
+        Locale match = Locale.lookup(ranges, locales);
+        assertNull(match, "[Locale.lookup failed on language"
+                + " range: " + ranges + " and language tags "
+                + locales + "]");
```
[the remark this answers is not in the document]
Comment (justin-curtis-lu) [reply test/jdk/java/util/Locale/Bug8135061.java]: I did a second take before pushing and mistakenly removed those blocks, I knew I had left them in there originally for a reason. Fixed.

**Review by naotoj — APPROVED** [review]
LGTM

**Pull request closed without merging**